arq/worker.py

```diff
@@ -757,8 +764,30 @@ def _add_signal_handler(self, signum: Signals, handler: Callable[[Signals], None
     def _jobs_started(self) -> int:
         return self.jobs_complete + self.jobs_retried + self.jobs_failed + len(self.tasks)
 
+    async def _wait_for_tasks_to_complete(self) -> None:
+        while self.tasks:
+            await asyncio.sleep(0.1)
+
     def handle_sig(self, signum: Signals) -> None:
         sig = Signals(signum)
+        if self._wait_for_job_completion_on_signal_seconds:
+            logger.info('Setting allow_pick_jobs to `False`')
+            self.allow_pick_jobs = False
+            logger.info(
+                'shutdown on %s ◆ %d jobs complete ◆ %d failed ◆ %d retries ◆ %d ongoing waiting for completion',
```

Comment:
```suggestion
                'shutdown on %s ◆ %d jobs complete ◆ %d failed ◆ %d retries ◆ %d to be completed',
```
```suggestion
                'shutdown on %s ◆ %d jobs complete ◆ %d failed ◆ %d retries ◆ %d ongoing',
```
```suggestion
                'shutdown on %s ◆ %d jobs complete ◆ %d failed ◆ %d retries ◆ %d waiting for completion',
```

I am too tired to find a good sentence 😁 help

Reply:
`to be completed` LGTM, not committing to avoid messing with your work.